api-tests/inventory/agents_test.go

```diff
@@ -554,21 +554,23 @@ func TestQanAgentExporter(t *testing.T) {
 		}, getAgentRes)
 
 		// Test change API.
-		changeQANMySQLPerfSchemaAgentOK, err := client.Default.AgentsService.ChangeQANMySQLPerfSchemaAgent(
-			&agents.ChangeQANMySQLPerfSchemaAgentParams{
-				Body: agents.ChangeQANMySQLPerfSchemaAgentBody{
-					AgentID: agentID,
-					Common: &agents.ChangeQANMySQLPerfSchemaAgentParamsBodyCommon{
-						Disable:            true,
-						RemoveCustomLabels: true,
+		changeQANMySQLPerfSchemaAgentOK, err := client.Default.AgentsService.ChangeAgent(
+			&agents.ChangeAgentParams{
+				Body: agents.ChangeAgentBody{
+					QANMysqlPerfschemaAgent: &agents.ChangeAgentParamsBodyQANMysqlPerfschemaAgent{
+						AgentID: agentID,
+						Common: &agents.ChangeAgentParamsBodyQANMysqlPerfschemaAgentCommon{
```

Comment:
can we move common fields one level up to reuse the same code?

Reply:
I created a PR to cover that, but it seems like the benefit is not obvious. https://perconadev.atlassian.net/browse/PMM-12875
Let's discuss offline.

Reply:
Decided to close that one as it may be difficult to refactor such a payload structure.